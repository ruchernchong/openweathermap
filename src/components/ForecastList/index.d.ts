import * as React from "react";

declare const ForecastList: React.ComponentType;

export default ForecastList;
