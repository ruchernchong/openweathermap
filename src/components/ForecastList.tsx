import * as React from "react";
import { useCallback } from "react";
import { WeatherCard } from "./WeatherCard";

interface TempRange {
  day: number;
  min: number;
  max: number;
  night: number;
  eve: number;
  morn: number;
}

interface Weather {
  id: number;
  description: string;
  icon: string;
  main: string;
}

interface Daily {
  dt: number;
  humidity: number;
  sunrise: number;
  sunset: number;
  temp: TempRange;
  weather: Weather[];
}

interface Hourly {
  dt: number;
  humidity: number;
  temp: number;
  weather: Weather[];
}

interface Props {
  daily: Daily[];
  hourly: Hourly[];
  isDailyForecast: boolean;
}

export const ForecastList: React.FC<Props> = ({
  daily,
  hourly,
  isDailyForecast
}) => {
  const renderForecast = useCallback(() => {
    if (isDailyForecast) {
      return daily.map(({ dt, humidity, sunrise, sunset, temp, weather }) => (
        <WeatherCard
          key={dt}
          dt={dt}
          humidity={humidity}
          sunrise={sunrise}
          sunset={sunset}
          tempRange={temp}
          weather={weather}
          isDailyForecast={isDailyForecast}
        />
      ));
    } else {
      return hourly.map(({ dt, humidity, temp, weather }) => (
        <WeatherCard
          key={dt}
          dt={dt}
          humidity={humidity}
          temp={temp}
          weather={weather}
          isDailyForecast={isDailyForecast}
        />
      ));
    }
  }, [isDailyForecast]);

  return <div data-testid="forecast-list">{renderForecast()}</div>;
};
