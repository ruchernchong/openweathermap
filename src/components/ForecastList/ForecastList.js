import React from "react";
import PropTypes from "prop-types";

import WeatherCard from "../WeatherCard";

export const ForecastList = ({ list }) => {
  return (
    <div data-testid="forecast-list">
      {list.map(({ dt, humidity, sunrise, sunset, temp, weather }) => (
        <WeatherCard
          key={dt}
          dt={dt}
          humidity={humidity}
          sunrise={sunrise}
          sunset={sunset}
          temp={temp}
          weather={weather}
        />
      ))}
    </div>
  );
};

ForecastList.propTypes = {
  list: PropTypes.array
};

export default ForecastList;
