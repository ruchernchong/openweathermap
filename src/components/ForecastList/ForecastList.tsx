import * as React from "react";

import WeatherCard from "../WeatherCard";

interface Temp {
  day: number;
  min: number;
  max: number;
  night: number;
  eve: number;
  morn: number;
}

interface Weather {
  id: number;
  description: string;
  icon: string;
  main: string;
}

interface List {
  dt: number;
  humidity: number;
  sunrise: number;
  sunset: number;
  temp: Temp;
  weather: Weather[];
}

interface Props {
  list: List[];
}

export const ForecastList: React.FC<Props> = ({ list }) => {
  return (
    <div data-testid="forecast-list">
      {list.map(({ dt, humidity, sunrise, sunset, temp, weather }) => (
        <WeatherCard
          key={dt}
          dt={dt}
          humidity={humidity}
          sunrise={sunrise}
          sunset={sunset}
          temp={temp}
          weather={weather}
        />
      ))}
    </div>
  );
};

export default ForecastList;
