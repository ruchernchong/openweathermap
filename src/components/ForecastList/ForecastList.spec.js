import React from "react";
import { render } from "@testing-library/react";
import ForecastList from "./ForecastList";

import mockForecast from "../../../__mocks__/forecast.json";

const { daily } = mockForecast.data;

describe("ForecastList", () => {
  test("should render", () => {
    const { container } = render(<ForecastList list={daily} />);
    expect(container).toMatchSnapshot();
  });
});
