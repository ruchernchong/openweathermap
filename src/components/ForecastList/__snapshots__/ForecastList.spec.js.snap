// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastList should render 1`] = `
<div>
  <div
    style="margin-bottom: 64px;"
  >
    <div>
      <h4>
        Sun Jun 28 2020 13:00:00 GMT+0800 (Singapore Standard Time)
      </h4>
      <img
        alt=""
        src="http://openweathermap.org/img/wn/10d@2x.png"
      />
      <div>
        Description: 
        light rain
      </div>
      <div>
        High: 
        30.7°C
      </div>
      <div>
        Low: 
        27.7°C
      </div>
      <button>
        Click more...
      </button>
    </div>
  </div>
  <div
    style="margin-bottom: 64px;"
  >
    <div>
      <h4>
        Mon Jun 29 2020 13:00:00 GMT+0800 (Singapore Standard Time)
      </h4>
      <img
        alt=""
        src="http://openweathermap.org/img/wn/10d@2x.png"
      />
      <div>
        Description: 
        light rain
      </div>
      <div>
        High: 
        30.0°C
      </div>
      <div>
        Low: 
        27.1°C
      </div>
      <button>
        Click more...
      </button>
    </div>
  </div>
  <div
    style="margin-bottom: 64px;"
  >
    <div>
      <h4>
        Tue Jun 30 2020 13:00:00 GMT+0800 (Singapore Standard Time)
      </h4>
      <img
        alt=""
        src="http://openweathermap.org/img/wn/10d@2x.png"
      />
      <div>
        Description: 
        moderate rain
      </div>
      <div>
        High: 
        29.8°C
      </div>
      <div>
        Low: 
        27.0°C
      </div>
      <button>
        Click more...
      </button>
    </div>
  </div>
  <div
    style="margin-bottom: 64px;"
  >
    <div>
      <h4>
        Wed Jul 01 2020 13:00:00 GMT+0800 (Singapore Standard Time)
      </h4>
      <img
        alt=""
        src="http://openweathermap.org/img/wn/10d@2x.png"
      />
      <div>
        Description: 
        light rain
      </div>
      <div>
        High: 
        30.0°C
      </div>
      <div>
        Low: 
        26.8°C
      </div>
      <button>
        Click more...
      </button>
    </div>
  </div>
  <div
    style="margin-bottom: 64px;"
  >
    <div>
      <h4>
        Thu Jul 02 2020 13:00:00 GMT+0800 (Singapore Standard Time)
      </h4>
      <img
        alt=""
        src="http://openweathermap.org/img/wn/10d@2x.png"
      />
      <div>
        Description: 
        light rain
      </div>
      <div>
        High: 
        30.0°C
      </div>
      <div>
        Low: 
        27.2°C
      </div>
      <button>
        Click more...
      </button>
    </div>
  </div>
  <div
    style="margin-bottom: 64px;"
  >
    <div>
      <h4>
        Fri Jul 03 2020 13:00:00 GMT+0800 (Singapore Standard Time)
      </h4>
      <img
        alt=""
        src="http://openweathermap.org/img/wn/10d@2x.png"
      />
      <div>
        Description: 
        light rain
      </div>
      <div>
        High: 
        29.9°C
      </div>
      <div>
        Low: 
        26.9°C
      </div>
      <button>
        Click more...
      </button>
    </div>
  </div>
  <div
    style="margin-bottom: 64px;"
  >
    <div>
      <h4>
        Sat Jul 04 2020 13:00:00 GMT+0800 (Singapore Standard Time)
      </h4>
      <img
        alt=""
        src="http://openweathermap.org/img/wn/10d@2x.png"
      />
      <div>
        Description: 
        moderate rain
      </div>
      <div>
        High: 
        29.9°C
      </div>
      <div>
        Low: 
        27.1°C
      </div>
      <button>
        Click more...
      </button>
    </div>
  </div>
  <div
    style="margin-bottom: 64px;"
  >
    <div>
      <h4>
        Sun Jul 05 2020 13:00:00 GMT+0800 (Singapore Standard Time)
      </h4>
      <img
        alt=""
        src="http://openweathermap.org/img/wn/10d@2x.png"
      />
      <div>
        Description: 
        moderate rain
      </div>
      <div>
        High: 
        30.3°C
      </div>
      <div>
        Low: 
        26.9°C
      </div>
      <button>
        Click more...
      </button>
    </div>
  </div>
  <div
    style="margin-bottom: 64px;"
  >
    <div>
      <h4>
        Mon Jul 06 2020 13:00:00 GMT+0800 (Singapore Standard Time)
      </h4>
      <img
        alt=""
        src="http://openweathermap.org/img/wn/10d@2x.png"
      />
      <div>
        Description: 
        light rain
      </div>
      <div>
        High: 
        30.2°C
      </div>
      <div>
        Low: 
        26.8°C
      </div>
      <button>
        Click more...
      </button>
    </div>
  </div>
  <div
    style="margin-bottom: 64px;"
  >
    <div>
      <h4>
        Tue Jul 07 2020 13:00:00 GMT+0800 (Singapore Standard Time)
      </h4>
      <img
        alt=""
        src="http://openweathermap.org/img/wn/04d@2x.png"
      />
      <div>
        Description: 
        broken clouds
      </div>
      <div>
        High: 
        30.4°C
      </div>
      <div>
        Low: 
        26.9°C
      </div>
      <button>
        Click more...
      </button>
    </div>
  </div>
  <div
    style="margin-bottom: 64px;"
  >
    <div>
      <h4>
        Wed Jul 08 2020 13:00:00 GMT+0800 (Singapore Standard Time)
      </h4>
      <img
        alt=""
        src="http://openweathermap.org/img/wn/10d@2x.png"
      />
      <div>
        Description: 
        light rain
      </div>
      <div>
        High: 
        30.8°C
      </div>
      <div>
        Low: 
        27.2°C
      </div>
      <button>
        Click more...
      </button>
    </div>
  </div>
  <div
    style="margin-bottom: 64px;"
  >
    <div>
      <h4>
        Thu Jul 09 2020 13:00:00 GMT+0800 (Singapore Standard Time)
      </h4>
      <img
        alt=""
        src="http://openweathermap.org/img/wn/10d@2x.png"
      />
      <div>
        Description: 
        light rain
      </div>
      <div>
        High: 
        29.8°C
      </div>
      <div>
        Low: 
        26.9°C
      </div>
      <button>
        Click more...
      </button>
    </div>
  </div>
  <div
    style="margin-bottom: 64px;"
  >
    <div>
      <h4>
        Fri Jul 10 2020 13:00:00 GMT+0800 (Singapore Standard Time)
      </h4>
      <img
        alt=""
        src="http://openweathermap.org/img/wn/10d@2x.png"
      />
      <div>
        Description: 
        moderate rain
      </div>
      <div>
        High: 
        30.0°C
      </div>
      <div>
        Low: 
        26.9°C
      </div>
      <button>
        Click more...
      </button>
    </div>
  </div>
  <div
    style="margin-bottom: 64px;"
  >
    <div>
      <h4>
        Sat Jul 11 2020 13:00:00 GMT+0800 (Singapore Standard Time)
      </h4>
      <img
        alt=""
        src="http://openweathermap.org/img/wn/10d@2x.png"
      />
      <div>
        Description: 
        heavy intensity rain
      </div>
      <div>
        High: 
        28.7°C
      </div>
      <div>
        Low: 
        26.3°C
      </div>
      <button>
        Click more...
      </button>
    </div>
  </div>
</div>
`;
